File: Python/src/leetcode_003.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if s == "":
            return 0
        if len(s) == 1:
            return 1
        max_len = 0
        for i in range(len(s)):
            for j in range(i+1, len(s)+1):
                if len(set(s[i:j])) == len(s[i:j]):
                    max_len = max(max_len, len(s[i:j]))
        return max_len

File: C/src/leetcode_003.c
#include <string.h>

int lengthOfLongestSubstring(char *s) {
  int len = strlen(s);
  int max = 0;
  int l = 0;
  int r = 0;

  while (r < len) {
    for (int i = l; i < r; i++) {
      if (s[i] == s[r]) {
        l = i + 1;
        break;
      }
    }
    if (r - l + 1 > max) {
      max = r - l + 1;
    }
    r++;
  }

  return max;
}
